### DISCUSSION TITLE
Cannot build jinx-mod on Fedora

### DISCUSSION THREAD

**Discussion by nivaca** [minad/jinx#194] [Q&A]
I cannot build `jinx-mod` on Fedora 40. I have installed (as I have done before) `enchant2` and `enchant2-devel`. 

```sh
❯ sudo dnf install -y enchant2 enchant2-devel
[sudo] password for nivaca: 
Updating and loading repositories:
Repositories loaded.
Package "enchant2-2.6.9-1.fc40.x86_64" is already installed.
Package "enchant2-2.6.9-1.fc40.i686" is already installed.
Package "enchant2-devel-2.6.9-1.fc40.i686" is already installed.
```

When I try to build it manually, I get:

```sh
~/.emacs.d/straight/repos/jinx 
❯ cc -I. -O2 -Wall -Wextra -fPIC -shared -o jinx-mod.so jinx-mod.c -I/usr/include/enchant-2 -lenchant-2
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-redhat-linux/14/../../../libenchant-2.so when searching for -lenchant-2
/usr/bin/ld: skipping incompatible /lib/libenchant-2.so when searching for -lenchant-2
/usr/bin/ld: skipping incompatible /usr/lib/libenchant-2.so when searching for -lenchant-2
/usr/bin/ld: cannot find -lenchant-2: No such file or directory
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-redhat-linux/14/../../../libenchant-2.so when searching for -lenchant-2
/usr/bin/ld: skipping incompatible /lib/libenchant-2.so when searching for -lenchant-2
/usr/bin/ld: skipping incompatible /usr/lib/libenchant-2.so when searching for -lenchant-2
collect2: error: ld returned 1 exit status
```

My system details are:
- Operating System: Fedora Linux 40
- KDE Plasma Version: 6.1.1
- KDE Frameworks Version: 6.3.0
- Qt Version: 6.7.2
- Kernel Version: 6.9.7-200.fc40.x86_64 (64-bit)
- Graphics Platform: Wayland

Any ideas on how to solve this? Thanks in advance.

**Comment by minad** [comment]
As far as I know the message "skipping incompatible ..." can appear if you use a mixed system where some libraries are 32 bit and some are 64 bit, when your compiler tries to compile a 32 bit version but your libraries are 64 bit or vice versa.

**Comment by nivaca** [comment]
This is the first time this happens. AFAIK I have no 32 bit libraries installed, only 64 bit ones. I manually installed the latest libenchant (2.8.1) with the same result.

**Reply by minad** [reply]
I wonder if you have found the solution to the problem by now. I don't have much Fedora experience so it is difficult for me to diagnose.

**Reply by nivaca** [reply]
No. I had to manually compile it and install it myself (it's pretty straightforward, to be sure).

**Reply by Thaodan** [reply]
I wonder if that is just the the linker being verbose. You could also reuse the openSUSE spec file to build Jinx for Fedora.

https://build.opensuse.org/package/show/editors/emacs-jinx

**Reply by mavit** [reply]
> AFAIK I have no 32 bit libraries installed

Architecture `i686` is 32 bit; `x86_64` is 64 bit.  These are the 32 bit libraries:

```
Package "enchant2-2.6.9-1.fc40.i686" is already installed.
Package "enchant2-devel-2.6.9-1.fc40.i686" is already installed.
```

I suggest either:

```bash
sudo dnf install enchant2-devel.x86_64
```

Or:
```bash
sudo dnf swap enchant2-devel.i686 enchant2-devel.x86_64
```
